### ISSUE TITLE
Add Language links + language assistance link to header

### ISSUE THREAD

**Issue by randyabramson** [18F/crt-portal#910]
### Background
We are being asked to make the language translations for the Reporting Portal more visible on the site. Our 'compromise' solution is to add the languages (and language assistance text) to the header of the site. For mobile, the language links will be inside the menu area.

### Assumptions
* The new language listings will be at the top of all Portal pages - the home page, all reporting pages, the privacy policy and error pages
* Language listing and language assistance text will be visible on desktop. For mobile, languages and language assistance links will be listed in the MENU area SEE MOCK IN COMMENTS); If languages  fit on tablet, keep it in header. If not, move to MENU.

### Question for DEVS
1. On Aviva's mobile mock, the 'on touch' state, Aviva shows a shaded area for the touched selection. Note that this is the current experience in the MENU now. Can we extend that function for the new languages.

### Reference
The full lineup of languages is below. Note that the identifiers in the parenthesis is for our own reference and will not be listed on the site.

简体字 (Chinese Simplified)
簡體字 Chinese Traditional)
English
Español (Spanish)
한국어 (Korean)
Tagalog
Tiếng Việt (vietnamese)

### Acceptance criteria for Sprint 41 

!! For Sprint 41, we will only be focusing on removing the language selector drop down and putting language links for English and Spanish + language assistance link in the header. 

1) Given I am on the homepage, any page of the form, privacy page or error page
when I look at the top of the page
then I see the globe icon and a list of available translation languages and the text 'Language Assistance Services' (see mock below)

2) Given I am on the homepage, any page of the form, privacy page or error page
when I mouseover a language link
then an underline appears under that respective language link

3) Given I am on the homepage, any page of the form, privacy page or error page
when I click on a specific language link
then the entire page is translated into that language

4) Given I am on the homepage, any page of the form, privacy page or error page
when I click the ''Availability of Language Assistance Services' link
then I go to https://www.justice.gov/crt/availability-language-assistance-services

![language list aviva mock top.png](https://images.zenhubusercontent.com/5ef156d51dd4f26e61414bbf/ad6b870f-9705-4a83-b2b9-307dc09c3fdd)

**Comment by randyabramson** [comment]
Adding Aviva's mobile mock for languages here


![mobile-translation-mock-aviva-032321.PNG](https://images.zenhubusercontent.com/5ef156d51dd4f26e61414bbf/8d774ce4-4d17-4016-9ab7-64e88723946f)

**Comment by ilodidoj** [comment]
@randyabramson a mock of the text 'Language Assistance Services'  will be helpful for the dev team on how that should display on the top header bar. Please confirm on removing the bottom footer language GO button.  We are in favor of keeping that as is, because this gives user to select language, if they ever go to the bottom of the page.

**Comment by randyabramson** [comment]
@ilodidoj 

1. The top area is getting crowded, so let's leave the 'language assistance services' link for the mobile menu and the bottom of the page for now. So, in short, the desktop view will not have the language assistance services' link in the header.

2. I am not following what you are mentioning about the 'go' button. Can you rephrase? 

Our view: We are in favor of keeping the bottom drop down selector. It was suggested that we drop the "GO" button and just have the language change when the user selects the language. But we are open to other ideas...we can be easily persuaded if you have another point of view.

cc: @blaughman @timoballard @AvivaOskow @emilycryan

**Comment by emilycryan** [comment]
@ilodidoj We wanted to see if it was feasible to remove the GO button and have the page auto-change to the selected language without the user having to select the language and then click the GO button. So:

- user scrolls to page bottom;
- user selects Spanish;
- page automatically changes to spanish immediately;
- page reloads and the user is at the top;

No need to push GO. CAVEAT: I'm assuming this is ok from an accessibility standpoint but if not, we would need to keep the button. 

CC @randyabramson @blaughman @timoballard @AvivaOskow

**Comment by randyabramson** [comment]
Thanks for the above @emilycryan 

Here's WW3's guidance on menus:
https://www.w3.org/WAI/tutorials/menus/flyout/

There is mention of using a 'button,' but it is more about using the button to activate the dropdown menu.

TBH, I can see both sides from an accessibility standpoint:
* With button: we are making them jump through an additional hoop
* Without button: if you accidentally select a language, now you have a screen reader speaking to you in a language you don't understand.

SO...I say that maybe we leave the button there for now and we can talk to Terri on Thursday when she is in the office. In short, let's not hold up development.

@emilycryan @timoballard @blaughman @ilodidoj

**Comment by AvivaOskow** [comment]
## Desktop annotations

### Font size: 14px

![Screen Shot 2021-03-26 at 1.06.02 PM.png](https://images.zenhubusercontent.com/5846f186b9dc3c6220230bf1/3ecc00f8-453b-4673-a221-338267b7754b)

**Comment by randyabramson** [comment]
Thank you @AvivaOskow cc: @timoballard @ilodidoj   ^^^^

**Comment by AvivaOskow** [comment]
# Mobile annotations

![Screen Shot 2021-03-26 at 3.19.22 PM.png](https://images.zenhubusercontent.com/5846f186b9dc3c6220230bf1/48b3b59c-8e7e-4e9a-b00a-0c694216b1f0)

![Screen Shot 2021-03-26 at 3.19.31 PM.png](https://images.zenhubusercontent.com/5846f186b9dc3c6220230bf1/87a2f4e5-e66a-4b32-8e79-7619b004a6ba)

**Comment by ilodidoj** [comment]
@randyabramson @blaughman @AvivaOskow @emilycryan Just FYI, mobile menu is not visible in current production, staging and dev when you are in the report data input page (any steps).  Menu is available only in landing page and next page before going to actual report data entry using "Submit a Report" button.  Users option to change language is to use the footer language change options from the data entry pages.

**Comment by randyabramson** [comment]
thank you @ilodidoj - please note that I put a note in SLACK that the footer disappears on step 2 (primary concern) on dev/staging/prod. 

Not sure if this is intentional. If not...it would be a great time to fix this as we move forward with translations.

@timoballard @blaughman @AvivaOskow @Jkrzy

**Comment by randyabramson** [comment]
@timoballard @ilodidoj - please know that the accessibility group suggested we keep the GO button on the language drop down in the footer.

Thanks!

cc @emilycryan @blaughman

**Comment by blaughman** [comment]
Function works as intended in Chrome and Edge.  Moving to UAT.

**Comment by Jkrzy** [comment]
There's a bit of new text here that will need translations, a variation of the footer language we have already translated.
>More Language Assistance Services

We have the assistance text in the footer translated already for Spanish.
>msgid "Availability of Language Assistance Services"
>msgstr "Disponibilidad de servicios de asistencia lingüística"

**Comment by randyabramson** [comment]
Thanks for catching that, @Jkrzy 

For now, can we go with translations for "Availability of Language Assistance Services,' since we have them? 

(I realize this text is longer, but we have a few big translation asks out with the FBI at the moment, some of them are still in motion).

Objections from @AvivaOskow @emilycryan ?

cc: @timoballard

**Comment by emilycryan** [comment]
Also one more thing - do we need to get translations for the GO button on the selector? I want to make sure if we are translating everything on the page, we don't lose that (esp for those using screen readers in alternative languages) Want to make sure we don't forget that little snippet. 

cc @randyabramson @Jkrzy @AvivaOskow @blaughman

**Comment by randyabramson** [comment]
@emilycryan good point. We definitely asked for translations for the names of the languages in the drop down. As for the word "GO," we did NOT translate that for Spanish. We can ask the FBI for guidance there.

**Comment by Jkrzy** [comment]
@emilycryan, **definitely** that's a bug where 'Go' wasn't being translated at all. I've included a fix for that here https://github.com/usdoj-crt/crt-portal/pull/853.

While we're missing __portal specific__ translations for "Go", the application will pick-up the translations for "Go" in languages that Django provides. This includes Spanish, Vietnamese, Chinese (Simplified), and Chinese (Traditional).

**Comment by randyabramson** [comment]
Works as expected on Desktop and Mobile. Moving to Staging.

**Comment by randyabramson** [comment]
Works as expected on Desktop and Mobile. Moving to Closed.
